benchmark/run.py

```diff
@@ -273,6 +325,23 @@ def run_from_argv() -> None:
         default=Path("/tmp/gpflow_benchmark_data"),
         help="Where to cache stuff, like datasets.",
     )
+    parser.add_argument(
+        "--shard",
+        default=ShardingStrategy("no"),
+        type=ShardingStrategy,
+        help=(
+            "Sharding strategy:"
+            " If set to 'no' this script performs all necessary work."
+            " If set to 'start' this script creates an output directory for the shards, and"
+            " prints its path."
+            " If set to the format '<i>/<n>', where 0 <= i < n then this script only runs"
+            " benchmarks for shard 'i' out of 'n', and stores partial results in 'dest'."
+            " This requires '--no_subdir'."
```

Comment:
> Why is no_subdir required for i/n and collect? Just to make sure that they're compatible with the previously run start?

Well we're reading / writing to the same directory, so I don't want to have it automatically create new directories. I could do that "automatically", but I think it will be more consistent and less surprising if we do things explicitly.